conseil-common/src/main/scala/tech/cryptonomic/conseil/common/tezos/TezosTypes.scala

```diff
@@ -2,6 +2,7 @@ package tech.cryptonomic.conseil.common.tezos
 
 import java.time.{Instant, ZonedDateTime}
 import scala.util.Try
+import tech.cryptonomic.conseil.common.generic.chain.DataTypes.BlockHash
```

Comment:
I'm actually quite worried by this choice.

Sharing data models was one of the major points for DDD to be born... is this really useful?

Reply:
I'm not sure why you are worried? I can copy this into two different places, but then, we will have `TezosBlockHash` and `BitcoinBlockHash`. At least I'm using this case class while implementing API.

Reply:
Let's keep it like this, for now.
I'm worried because different domains should have different data-types if they can evolve independently (Bounded Contexts).
It's not a big thing with a single definition. But doing this could give the impression for any future developer, that re-using the code like this in different domains is a goal.
I can cope with this choice but I'm not sure I like it.

Reply:
I understand. Then, BlockHash cannot be in `generic` package. If I understand your way of thinking correctly. Both for me works, so I can duplicate that part, if that will work better for you :)

Reply:
```
==================
conseil-lorre_1     | 14:04:46.510 [lorre-tezos-indexer-akka.actor.default-dispatcher-3] INFO  t.c.c.indexer.tezos.TezosIndexer     - Completed processing 0.19% of total requested blocks
conseil-lorre_1     | 14:04:46.510 [lorre-tezos-indexer-akka.actor.default-dispatcher-3] INFO  t.c.c.indexer.tezos.TezosIndexer     - Estimated time to finish is around 1247 minutes
conseil-lorre_1     | 14:04:46.510 [lorre-tezos-indexer-akka.actor.default-dispatcher-3] INFO  t.c.c.indexer.tezos.TezosIndexer     - =====================================================================================
```

and `localhost/v2/data/tezos/mainnet/blocks/BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY`
returned:
```
{
    "block": {
        "level": 17,
        "proto": 1,
        "predecessor": "BMSJNTcywR8CwQm8VPK5y4hDdsFsDHhqfW77JcnZRJ3UZXtfYDj",
        "timestamp": 1575020755000,
        "fitness": "01,0000000000000010",
        "context": "CoVmAGaK27JwMk8arfvBKFkrxaT6SBHbKVCsk2ipfPmrXmoYroUA",
        "signature": "sigsVSdkVywDgkQ8aaZqHwt8PRCgCmR7PyVYsUeTQiBZyhqtjuonUkabsMsLyfAjYfUySDZrNbeRDajxKXUxyBbeTtVon6gV",
        "protocol": "PsBabyM1eUXZseaJdmXFApDSBqj8YBfwELoxZHHW77EMcAbbwAS",
        "chainId": "NetXjD3HPJJjmcd",
        "hash": "BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY",
        "operationsHash": "LLoZU7ykB5tEiaGThwx3yNQnmoZyWvbMEGDYWnoQiY3hw8qsPuRXb",
        "periodKind": "proposal",
        "currentExpectedQuorum": 5800,
        "baker": "tz1RomaiWJV3NFDZWTMVR2aEeHknsn3iF5Gi",
        "consumedGas": 0,
        "metaLevel": 17,
        "metaLevelPosition": 16,
        "metaCycle": 0,
        "metaCyclePosition": 16,
        "metaVotingPeriod": 0,
        "metaVotingPeriodPosition": 16,
        "priority": 1,
        "utcYear": 2019,
        "utcMonth": 11,
        "utcDay": 29,
        "utcTime": "09:45:55"
    },
    "operation_groups": [
        {
            "protocol": "PsBabyM1eUXZseaJdmXFApDSBqj8YBfwELoxZHHW77EMcAbbwAS",
            "chainId": "NetXjD3HPJJjmcd",
            "hash": "opVy917c5L2rxTNGKDtzUDtdH5sucpxtmyMN8DRWy1aRdKfVBn6",
            "branch": "BMSJNTcywR8CwQm8VPK5y4hDdsFsDHhqfW77JcnZRJ3UZXtfYDj",
            "signature": "sigqQqrwzZ1nvBXdCWe9dpGachXMoBm56YyRje8BQ9a76ogQs5fodBSWEbACrijs6x1EzGyDw4LAGTiFWSCXMxktPsCeBYJ2",
            "blockId": "BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY",
            "blockLevel": 17
        },
        {
            "protocol": "PsBabyM1eUXZseaJdmXFApDSBqj8YBfwELoxZHHW77EMcAbbwAS",
            "chainId": "NetXjD3HPJJjmcd",
            "hash": "ooRycnFvJuL7qv2ZCy6jcFTjhg86Rb8xYRsLxkqpa5eXhoR1Wzd",
            "branch": "BMSJNTcywR8CwQm8VPK5y4hDdsFsDHhqfW77JcnZRJ3UZXtfYDj",
            "signature": "sigtwN5KZWjpep4jNW8UpWhntESmLoCcFjfBeSi5qyKWJ7f2KRMXsVn1hySHS8qSUqxW3hMuEwfczcvzp2EdYgScEzZD3L5q",
            "blockId": "BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY",
            "blockLevel": 17
        },
        {
            "protocol": "PsBabyM1eUXZseaJdmXFApDSBqj8YBfwELoxZHHW77EMcAbbwAS",
            "chainId": "NetXjD3HPJJjmcd",
            "hash": "oo8y7V31U4vAaJ1FNwaY75eE1YdUg3BKjXtbeW6jFoeJbGfwXVW",
            "branch": "BMSJNTcywR8CwQm8VPK5y4hDdsFsDHhqfW77JcnZRJ3UZXtfYDj",
            "signature": "signfDxMUwitg43QCxTgUZUMnUzvmfkaJWC5Atfy1dcgxDxMT9g9WwovCjPJU3pUYGypiWdjSWN6MZpMpghtkbfBWVkjWAEd",
            "blockId": "BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY",
            "blockLevel": 17
        },
        {
            "protocol": "PsBabyM1eUXZseaJdmXFApDSBqj8YBfwELoxZHHW77EMcAbbwAS",
            "chainId": "NetXjD3HPJJjmcd",
            "hash": "opNhNyaD8w9E7hu68Jq3veiyvPGjeDdgufyVmG1ib1ZKxNx2WL5",
            "branch": "BMSJNTcywR8CwQm8VPK5y4hDdsFsDHhqfW77JcnZRJ3UZXtfYDj",
            "signature": "sigtkHe3saXSJDxVGy9kn79krSjFBCGanYNDUnC3Gfc37nDDkFuA1tbFDbAGDYiu8piMddoWPoURHbp7w1DiAMdLqk3v7eeS",
            "blockId": "BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY",
            "blockLevel": 17
        },
        {
            "protocol": "PsBabyM1eUXZseaJdmXFApDSBqj8YBfwELoxZHHW77EMcAbbwAS",
            "chainId": "NetXjD3HPJJjmcd",
            "hash": "opVeVGWwBJBvKCrqHvbGaDscB6MTinFLjKM1q9H3KYiLrGKUcyb",
            "branch": "BMSJNTcywR8CwQm8VPK5y4hDdsFsDHhqfW77JcnZRJ3UZXtfYDj",
            "signature": "sigu2BDWwaf6AqiXyGVeZVAf2SXigUDgr16La9u4j1noQLjCcXXJJDGztfif6CEgHn8Wpfyp1dyBr55xrfHe3qXwo5vjsQwx",
            "blockId": "BKs4hhB8TKFP9T4c4o1k5jYQZuLgGacthXXmAEsYC1tTE6JjhYY",
            "blockLevel": 17
        }
    ]
}
```
so I consider this as a working solution.

Reply:
Personally I'd feel better with that duplication